File: Data Structures & Algorithms/anagram-groups/submission-6.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        const res = {}
        for(const s of strs){
            const sorted_s = s.split('').sort().join();
            if(!res[sorted_s]){
                res[sorted_s] = []
            }
            res[sorted_s].push(s)
        }

        return Object.values(res)
    }
}

File: Data Structures & Algorithms/generate-parentheses/submission-6.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        res = []

        def bt(c,o,sub):
            if o == c == n:
                res.append("".join(sub[:]))
                return 
            if o < n:
                sub.append("(")
                bt(c,o+1,sub)
                sub.pop()
            if c < o:
                sub.append(")")
                bt(c+1,o,sub)
                sub.pop()
        
        bt(0,0,[])
            

        return res